internal/s3storage/s3storage.go

```diff
@@ -68,3 +163,65 @@ func (s *Service) CreateArtifact(artifactSpec artifact.Spec, md5 string) (string
 
 	return uploadURL, nil
 }
+
+func (s *Service) Close() error {
+	if s.sqsHandlerQuitChannel != nil {
+		close(s.sqsHandlerQuitChannel)
+	}
+	if s.sqsHandlerErrorChannel != nil {
+		err := <-s.sqsHandlerErrorChannel
+		return err
+	}
+	return nil
+}
+
+func (s *Service) startSQSHandler(handler func(msg string) error) {
+	log.Infof("starting SQS handler")
+	s.sqsHandlerQuitChannel = make(chan struct{})
+	s.sqsHandlerErrorChannel = make(chan error, 1)
+	go func() {
+		for {
+			select {
+			case _, ok := <-s.sqsHandlerQuitChannel:
+				if !ok {
+					s.sqsHandlerErrorChannel <- nil
+					return
+				}
+			default:
+				// continue
+			}
+
+			output, err := s.sqsClient.ReceiveMessage(&sqs.ReceiveMessageInput{
+				QueueUrl:            aws.String(s.sqsQueueURL),
+				MaxNumberOfMessages: aws.Int64(1),
+				WaitTimeSeconds:     aws.Int64(1),
+			})
+
+			if err != nil {
+				log.Errorf("failed to fetch sqs messages %v", err)
+				continue
+			}
+
+			for _, message := range output.Messages {
+				err := handler(*message.Body)
+				if err != nil {
+					log.With("messageID", *message.MessageId, "messageBody", *message.Body).Errorf("Failed handling SQS message. Error: %s", err)
```

Comment:
Is it intentional that we drop messages that could not be handled? A missing `continue` here results in `DeleteMessage` being called on every message.

Reply:
Yeah, it was actually intentional, but I'm unsure if its a good idea. I'll add the `continue`